app/controllers/api/v1/materials_controller.rb

```diff
@@ -221,6 +236,44 @@ def get_favorites
 
   end
 
+  #
+  #
+  # Get a single materials item and return a json representation
+  # GET /api/v1/materials/get_material
+  #
+  def get_material
```

Comment:
The typical rails route for this would be `/api/v1/materials/:id`.  And the action name would be `show`.  However since there are both interactives and external activities that need to be returned it can't be this simple.  So I guess I'd do: `/api/v1/materials/:type/:id` And type should be lower case with underscores.